src/mlpack/tests/random_forest_test.cpp

```diff
@@ -1,609 +1,790 @@
+  // observation 1 has already been drawn.
+  SequentialBootstrap<>::ComputeNextDrawProbabilities(
+      phi1, 1, concurrency, invConcurrency, indM, delta2);
+
+  REQUIRE(delta2(0) == Approx(5.0 / 14.0));
+  // Should have the lowest probability as it has already been drawn.
+  REQUIRE(delta2(1) == Approx(3.0 / 14.0));
+  // Should have the highest probability as this event does not overlap
+  // with others.
+  REQUIRE(delta2(2) == Approx(6.0 / 14.0));
+}
+
+/**
+ * Test ComputeSamples.
+ */
+TEST_CASE("ComputeSamplesTest", "[RandomForestTest]")
+{
+  arma::umat indM(2 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  indM(1, 3) = indM(1, 4) = indM(1, 5) = 5;
+
+  SequentialBootstrap bootstrap(indM);
+  const arma::uvec phi(bootstrap.ComputeSamples(6));
+
+  // Can only check that the next draw does not yield indices
+  // outside of the range of observations.
+  REQUIRE(phi(0) < 6);
+  REQUIRE(phi(1) < 6);
+  REQUIRE(phi(2) < 6);
+}
+
+/**
+ * Test SequentialBootstrap.
+ */
+TEST_CASE("SequentialBootstrapTest", "[RandomForestTest]")
+{
+  const arma::mat ds(10 /* rows */, 6 /* cols */, arma::fill::randu);
+  const arma::Row<size_t> labels{ 1, 0, 0, 0, 0, 0 };
+  const arma::rowvec weights(6, arma::fill::ones);
+  arma::umat indM(2 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  indM(1, 3) = indM(1, 4) = indM(1, 5) = 5;
+
+  SequentialBootstrap<> bootstrap(indM);
+  arma::mat bsDataset;
+  arma::Row<size_t> bsLabels;
+  arma::rowvec bsWeights;
+
+  bootstrap.Bootstrap<true>(
+      ds, labels, weights, bsDataset, bsLabels, bsWeights);
+
+  // Check that dimensions are the same.
+  REQUIRE(ds.n_rows == bsDataset.n_rows);
+  REQUIRE(ds.n_cols == bsDataset.n_cols);
+  REQUIRE(labels.n_cols == bsLabels.n_cols);
+  REQUIRE(weights.n_cols == bsWeights.n_cols);
+}
+
+TEST_CASE("RandomForestWithSequentialBootstrapTest", "[RandomForestTest]")
+{
+  const arma::mat ds(10 /* rows */, 6 /* cols */, arma::fill::randu);
+  const arma::Row<size_t> labels{ 1, 0, 0, 0, 0, 0 };
+  const arma::rowvec weights(6, arma::fill::ones);
+  arma::umat indM(2 /* rows */, 6 /* cols */, arma::fill::zeros);
+
+  indM(0, 0) = 0;
+  indM(1, 0) = 2;
+  indM(0, 1) = 2;
+  indM(1, 1) = 3;
+  indM(0, 2) = 4;
+  indM(1, 2) = 5;
+
+  indM(1, 3) = indM(1, 4) = indM(1, 5) = 5;
+
+  SequentialBootstrap<> bootstrap(indM);
+
+  using RF = RandomForest<
+      mlpack::GiniGain,
+      mlpack::MultipleRandomDimensionSelect,
+      mlpack::BestBinaryNumericSplit,
+      mlpack::AllCategoricalSplit,
+      true,
+      mlpack::SequentialBootstrap<>>;
+
+  RF rf(
+      ds,
+      labels,
+      2,
+      weights,
+      20,
+      1,
+      1e-7,
+      0,
+      mlpack::MultipleRandomDimensionSelect(),
+      bootstrap);

```

Comment:
```suggestion
      bootstrap);

  // Sanity check: ensure that the predictions are reasonable.
  arma::Row<size_t> predictions;
  rf.Predict(ds, predictions);
  
  for (size_t i = 0; i < predictions.n_elem; ++i)
    REQUIRE(predictions[i] == 0 || predictions[i] == 1);
```

Just a sanity check here to actually use the random forest.  This doesn't check accuracy, just that the predictions are even close to correct.  This would just check the internal structure of the random forest---for instance, if somehow no trees were trained (e.g. if the bootstrap dataset was empty, for instance), this would throw an error.